### ISSUE TITLE
Godot 2D Sprite resolution change freezes (5 seconds)

### ISSUE THREAD

**Issue by SmartySmart702** [godotengine/godot#47746]
Godot version:
3.2.3
3.3 RC 7
3.3 RC 8

OS/device including version:
Win 10 x64
GTX 1070 8GB
Driver: Nvidia Studio 461.92 (also happened before with Game Ready Driver)
GLES3
GLES2

Issue description:
Godot freezes for ~5 seconds while changing the resolution of a 2D Sprite.
Somehow if you set the x resolution to 512 and press tab you are still in the x resolution (instead of changing to the y resolution) and if you type in 512 again and press enter Godot freezes.
Sometimes it can happen that after a freezes the y resolution is e.g. 256 but if you click on it it's 512.

Steps to reproduce:
1) Create a new Project
2) Add a 2D Node and a Sprite
3) Change Sprite Scale to x:1.3 and y:0.9#
4) Under Texture, add (create) a new ImageTexture
5) Not set the x resolution to e.g. 256, press tab and set the y resolution also to 256 and press enter
6) Do step 5 again but with 512 instead of 256

**Comment by floppyhammer** [comment]
I could reproduce this. In system console it says:
> Godot Engine v3.3.rc8.official - https://godotengine.org
> OpenGL ES 2.0 Renderer: GeForce GTX 1080 Ti/PCIe/SSE2
> OpenGL ES Batching: ON
> 
> Failed method: EditorSpinSlider:_update_callback target ID: 102129
> Failed method: RichTextLabel:_update_callback target ID: 8631
> TOTAL BYTES: 4194288
> NULL count: 0
> CALL : 174757
> CALL _update_callback: 2
> CALL show_modal: 1
> CALL grab_focus: 1
> CALL select_all: 1
> ERROR: Message queue out of memory. Try increasing 'memory/limits/message_queue/max_size_kb' in project settings.
>    At: core/message_queue.cpp:55
> Failed method: ToolButton:_update_callback target ID: 8633
> TOTAL BYTES: 4194288
> NULL count: 0
> CALL : 174757
> CALL _update_callback: 2
> CALL show_modal: 1
> CALL grab_focus: 1
> CALL select_all: 1
> ERROR: Message queue out of memory. Try increasing 'memory/limits/message_queue/max_size_kb' in project settings.
>    At: core/message_queue.cpp:55
> Failed method: ToolButton:_update_minimum_size target ID: 8633
> TOTAL BYTES: 4194288
> NULL count: 0
> CALL : 174757
> CALL _update_callback: 2
> CALL show_modal: 1
> CALL grab_focus: 1
> CALL select_all: 1
> ERROR: Message queue out of memory. Try increasing 'memory/limits/message_queue/max_size_kb' in project settings.
>    At: core/message_queue.cpp:55
> TOTAL BYTES: 4194288
> NULL count: 0
> CALL : 174757
> CALL _update_callback: 2
> CALL show_modal: 1
> CALL grab_focus: 1
> CALL select_all: 1
> ERROR: Message queue out of memory. Try increasing 'memory/limits/message_queue/max_size_kb' in project settings.
>    At: core/message_queue.cpp:55
> Failed method: LineEdit:show_modal target ID: 102131
> TOTAL BYTES: 4194288
> NULL count: 0
> CALL : 174757
> CALL _update_callback: 2
> CALL show_modal: 1
> CALL grab_focus: 1
> CALL select_all: 1
> ERROR: Message queue out of memory. Try increasing 'memory/limits/message_queue/max_size_kb' in project settings.
>    At: core/message_queue.cpp:55
> Failed method: LineEdit:grab_focus target ID: 102131
> TOTAL BYTES: 4194288
> NULL count: 0
> CALL : 174757
> CALL _update_callback: 2
> CALL show_modal: 1
> CALL grab_focus: 1
> CALL select_all: 1
> ERROR: Message queue out of memory. Try increasing 'memory/limits/message_queue/max_size_kb' in project settings.
>    At: core/message_queue.cpp:55
> Failed method: LineEdit:select_all target ID: 102131
> TOTAL BYTES: 4194288
> NULL count: 0
> CALL : 174757
> CALL _update_callback: 2
> CALL show_modal: 1
> CALL grab_focus: 1
> CALL select_all: 1
> ERROR: Message queue out of memory. Try increasing 'memory/limits/message_queue/max_size_kb' in project settings.
>    At: core/message_queue.cpp:55

**Comment by floppyhammer** [comment]
Also, this happens

![sprite2d_freeze](https://user-images.githubusercontent.com/28705694/114254343-52285b80-99e1-11eb-9ec4-01e6d9ff3c87.gif)

**Comment by Calinou** [comment]
Maybe we should unexpose the size parameters in the StreamTexture inspector. I don't see a reason to expose it as modifying it will break things.

**Comment by SmartySmart702** [comment]
It happened again (Godot 3.3) but for the scale value of a Spirte so its not only related to the res values. I'm not sure but I think this happens most likely (or only?) when you press tap to change the next value.

**Comment by berarma** [comment]
I'm having this problem on multiple numeric fields, in project settings, nodes, etc. I click in the field, type a value, then hit tab and sometimes Godot and the whole desktop freezes. It's doesn't happen everytime. While frozen the CPU goes up, at least a few core are maxed up.

I'm using Godot 3.3 from Flatpak.

**Comment by Calinou** [comment]
@berarma Can you reproduce this with Godot 3.3 downloaded from [godotengine.org](https://godotengine.org/download/) (i.e. not in a Flatpak)?

**Comment by berarma** [comment]
@Calinou Yes, I've tried just now. By the third or fourth time I've changed the window size values in the project settings the same has happened. For the first few seconds only Godot freezes, then the whole desktop, and after a few more seconds everything resumes working.

Like I said this is happening consistently 2 or 3 times a day when editing other fields in nodes or resoruces for the last few days that I just started using Godot.
